modules/openapi-generator/src/test/java/org/openapitools/codegen/ruby/RubyClientCodegenTest.java

```diff
@@ -319,13 +319,7 @@ public void exampleRegexParameterValidationOAS3Test() {
         Assert.assertEquals(op.allParams.get(0).pattern, "/^pattern$/");
         // pattern_two_slashes '/^pattern$/i'
         Assert.assertEquals(op.allParams.get(1).pattern, "/^pattern$/i");
-        // pattern_one_slash_start '/^pattern$'
-        Assert.assertEquals(op.allParams.get(2).pattern, "/^pattern$/");
-        // pattern_one_slash_end '^pattern$/'
-        Assert.assertEquals(op.allParams.get(3).pattern, "/^pattern$/");
-        // pattern_one_slash_near_end '^pattern$/im'
-        Assert.assertEquals(op.allParams.get(4).pattern, "/^pattern$/im");
```

Comment:
Hi @meganemura. I put in the fix for a case we found at work.  The OpenAPI v3 spec supports regex that conforms to ECMA 262 regular expressions (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#schema-object) and The ECMA 262 standard says JS RegExp expressions are modeled on those in Perl 5 (https://www.ecma-international.org/ecma-262/5.1/#sec-15.10) and Perl 5 does indeed support \d (https://perldoc.perl.org/perlre.html#Character-Classes-and-other-Special-Escapes)

Reply:
I was trying to ensure that if forward slashes where not surrounding the inputted regex then they would be added to force the inputted text to be a treated as a regex literal... I wanted to prevent someone from putting in malicious code via `pattern`.  The Mustache template doesn't surround the regex with quotes anymore.

I was thinking we could put forward slashes in the Mustache template but then we might have an ending forward slash when we didn't want one i.e. `/^pattern$/i/`.

Reply:
@meganemura I read the code in `addRegularExpressionDelimiter` and I like it.  Thanks heaps for the refactor! 

@wing328 I am happy to have those 3 tests removed.  @meganemura is right, they are not correct regular expressions.

Reply:
@ggershoni Thanks for reviewing 👍